server/src/main/java/org/elasticsearch/common/logging/NodeNamePatternConverter.java

```diff
@@ -50,23 +49,31 @@ public static void setNodeName(String nodeName) {
     /**
      * Called by log4j2 to initialize this converter.
      */
-    public static NodeNamePatternConverter newInstance(final String[] options) {
+    public static PatternConverter newInstance(final String[] options) {
         if (options.length > 0) {
             throw new IllegalArgumentException("no options supported but options provided: "
                     + Arrays.toString(options));
         }
-        return new NodeNamePatternConverter(NODE_NAME.get());
+        return new NodeNamePatternConverter();
     }
 
-    private final String nodeName;
-
-    private NodeNamePatternConverter(String nodeName) {
+    private NodeNamePatternConverter() {
         super("NodeName", "node_name");
-        this.nodeName = nodeName;
     }
 
     @Override
     public void format(LogEvent event, StringBuilder toAppendTo) {
+        String nodeName = NODE_NAME.get();
+        if (nodeName == null) {
+            /*
+             * If node.name isn't explicitly configured in elasticsearch.yml
+             * we won't have it during bootstrap but the node name pattern
+             * is still in the log format so we stick *something* in there.
+             * This is ok because we don't generally log anything before
+             * we have the node name.
+             */
+            nodeName = "-------";
```

Comment:
maybe something like `notset` or `unknown` would be more obvious to users what is going on?

Reply:
yeah. I'll see if I can do that.